static/app/views/organizationStats/getReasonGroupName.ts

```diff
@@ -160,9 +106,21 @@ const invalidReasonsGroup: Record<string, DiscardReason[]> = {
   sampling: [DiscardReason.TRANSACTION_SAMPLED],
 };
 
-function getInvalidReasonGroupName(reason: DiscardReason): string {
+function getInvalidReasonGroupName(reason: string): string {
+  // The reason for 'too large' are of the from 'too_large:reason'
+  // We want to convert this to 'too_large_reason' as we later apply
+  // the following logic to the reason: startCase(reason.replace(/-|_/g, ' '))
+  // That is, we want to 'too_large: attachment' to be displayed as
+  // 'Too Large Attachment' in the UI. The one exception to that
+  // is 'too_large:unknown' which should just be 'Too Large'
+  if (reason.startsWith('too_large:')) {
+    if (reason === 'too_large:unknown') {
+      return DiscardReason.TOO_LARGE;
+    }
+    return reason.replace('too_large:', 'too_large_');
+  }
```

Comment:
We can do better, split once the separator `:` and then use the existing logic on the first part.

Reply:
Not sure what you mean with existing logic on the first part here since it is of the form "too_large" anyways? Or would you just want to split on the `:` regardless of what the prefix is and if the prefix is bad somehow make the entire thing return `internal` else do `prefix + postfix` ?

Reply:
Talked offline with @Dav1dde and we came up with a better solution.